# Descripción
Analsis de datos con Pandas (Core)

#Objetivo

El objetivo de este desafío es evaluar tus habilidades en el manejo de datos utilizando Pandas. Trabajarás con un dataset real extraído de Kaggle y aplicarás técnicas de carga, exploración, filtrado y slicing de datos.



# Dataset


Utilizaremos el dataset de «Video Game Sales» que contiene información sobre las ventas de videojuegos en diferentes regiones del mundo. El dataset incluye detalles sobre el nombre del juego, la plataforma, el año de lanzamiento, el género, el editor, y las ventas en diferentes regiones. Puedes descargar el dataset desde Kaggle.
https://www.kaggle.com/datasets/gregorut/videogamesales

# 1. Preparación del Entorno
- Asegúrate de tener instalado Pandas en tu entorno de trabajo.
- Descarga el archivo vgsales.csv desde Kaggle y colócalo en tu directorio de trabajo.

In [9]:
import pandas as pd

# 2. Cargar los Datos
- Carga el archivo vgsales.csv en un DataFrame de Pandas.
- Muestra las primeras 10 filas del DataFrame para confirmar que los datos se han cargado correctamente.

In [10]:
# Cargar datos desde un archivo CSV
df = pd.read_csv('/content/vgsales.csv')

# Mostrar las primeras 10 filas del DataFrame
print(df.head(10))

   Rank                       Name Platform    Year         Genre Publisher  \
0     1                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5     6                     Tetris       GB  1989.0        Puzzle  Nintendo   
6     7      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7     8                   Wii Play      Wii  2006.0          Misc  Nintendo   
8     9  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9    10                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29

# 3. Exploración Inicial de los Datos
- Muestra las últimas 5 filas del DataFrame.
- Utiliza el método info() para obtener información general sobre el DataFrame, incluyendo el número de entradas, nombres de las columnas, tipos de datos y - - memoria utilizada.
- Genera estadísticas descriptivas del DataFrame utilizando el método describe().

In [11]:
# Mostrar las últimas 5 filas del DataFrame
print("Últimas filas del DataFrame:")
print(df.tail(5))

# Obtener información general sobre el DataFrame
print("\n\n\nInformación del DataFrame:")
print(df.info())

# Generar estadísticas descriptivas
print("\n\n\nEstadísticas descriptivas del DataFrame:")
print(df.describe())


Últimas filas del DataFrame:
        Rank                                              Name Platform  \
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
16593  2002.0  Platform       Kemco      0.01      0.00       0.0   
16594  2003.0   Shooter  Infogrames      0.01      0.00       0.0   
16595  2008.0    Racing  Activision      0.00      0.00       0.0   
16596  2010.0    Puzzle    7G//AMES      0.00      0.01       0.0   
16597  2003.0  Platform     Wanadoo      0.01      0.00       0.0   

       Other_Sales  Global_Sales  
16593          0.0          0.01  
16594          0.0          0.01  


## Análisis

1. Hay 16598 entradas o datos, con índices de 0 to 16597
2. Columnas como "Year" y "Publisher" tiene datos nulos
3. Hay 1 columna tipo int ("Rank"), 4 columnas tipo object y 6 columnas son tipo float64
4. Al ser Ventas en millones de dólares, en Norte América, Europa, Japón y Otros, no es posible ver algún tipo de "tendencia" ya que todos los años están juntos, toda la información está "revuelta"

# 4. Inspección de los Datos
- Inspecciona los tipos de datos de cada columna utilizando el atributo dtypes.
- Cuenta los valores únicos en la columna Genre utilizando el método value_counts().
- Muestra todos los valores únicos en la columna Platform utilizando el método unique().

In [20]:
# Inspeccionar los tipos de datos de las columnas
print("Tipos de datos de las columnas:")
print(df.dtypes)

# Contar valores únicos en una columna Genre
print("\n\nConteo de valores únicos en la columna 'Genre':")
print(df['Genre'].value_counts())

# Obtener valores únicos en una columna Platform
print("\n\n\nValores únicos en la columna 'Platform':")
print(df['Platform'].unique())

# Me pareció más organizado verlo hacía abajo
print("\nValores únicos en la columna 'Platform' (ordenados alfabéticamente):")
print(pd.Series(sorted(df['Platform'].unique())))

Tipos de datos de las columnas:
Rank              int64
Name             object
Platform         object
Year            float64
Genre            object
Publisher        object
NA_Sales        float64
EU_Sales        float64
JP_Sales        float64
Other_Sales     float64
Global_Sales    float64
dtype: object


Conteo de valores únicos en la columna 'Genre':
Genre
Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
Adventure       1286
Racing          1249
Platform         886
Simulation       867
Fighting         848
Strategy         681
Puzzle           582
Name: count, dtype: int64



Valores únicos en la columna 'Platform':
['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' '3DS' 'PS4' 'N64'
 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'GC' 'WiiU' 'GEN' 'DC' 'PSV' 'SAT'
 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']

Valores únicos en la columna 'Platform' (ordenados alfabéticamente):
0     2600
1      3DO
2      3DS
3       DC
4       DS


# 5. Filtrado de Datos
- Filtra el DataFrame para mostrar solo las filas donde las ventas en América del Norte (NA_Sales) sean mayores a 1 millón.
- Filtra el DataFrame para mostrar solo las filas donde las ventas en Japón (JP_Sales) sean menores a 0.1 millón.
- Utilizando el método query(), filtra el DataFrame para mostrar las filas donde el género sea Action y las ventas globales (Global_Sales) sean mayores a 2 millones.

In [21]:
# Filtrar filas donde las ventas son mayores a 1 millón en América del Norte
na_sales_more_1m = df[(df['NA_Sales'] > 1)]
print(na_sales_more_1m)

      Rank                            Name Platform    Year         Genre  \
0        1                      Wii Sports      Wii  2006.0        Sports   
1        2               Super Mario Bros.      NES  1985.0      Platform   
2        3                  Mario Kart Wii      Wii  2008.0        Racing   
3        4               Wii Sports Resort      Wii  2009.0        Sports   
4        5        Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing   
...    ...                             ...      ...     ...           ...   
1905  1907                           Joust     2600  1982.0      Platform   
1917  1919                NCAA Football 14     X360  2013.0        Sports   
1953  1955                       NFL Blitz      N64  1998.0        Sports   
1954  1956         NFL Quarterback Club 98      N64  1997.0        Sports   
1957  1959  Diablo II: Lord of Destruction       PC  2001.0  Role-Playing   

                  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0

In [23]:
# Filtrar filas donde las ventas son mayores a 1 millón en América del Norte
jp_sales_lees_0_1m = df[(df['JP_Sales'] < 0.1)]
print(jp_sales_lees_0_1m)

        Rank                                              Name Platform  \
23        24                                Grand Theft Auto V     X360   
35        36                        Call of Duty: Black Ops II     X360   
36        37                    Call of Duty: Modern Warfare 2     X360   
60        61                                      Just Dance 3      Wii   
61        62                              Call of Duty: Ghosts     X360   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre             Publisher  NA_Sales  EU_Sales  JP_Sales  \
23     2013.0    Act

In [31]:
# Filtrar filas utilizando el método query género Action
query_genre_action_global_more_2m = df.query("Genre == 'Action' & Global_Sales > 2")
print(query_genre_action_global_more_2m)

     Rank                                               Name Platform    Year  \
16     17                                 Grand Theft Auto V      PS3  2013.0   
17     18                      Grand Theft Auto: San Andreas      PS2  2004.0   
23     24                                 Grand Theft Auto V     X360  2013.0   
24     25                        Grand Theft Auto: Vice City      PS2  2002.0   
38     39                               Grand Theft Auto III      PS2  2001.0   
..    ...                                                ...      ...     ...   
815   817             Tom Clancy's Splinter Cell: Conviction     X360  2010.0   
817   819                                    The Incredibles      GBA  2004.0   
833   835  Peter Jackson's King Kong: The Official Game o...      PS2  2005.0   
835   837                            Need for Speed: The Run      PS3  2011.0   
840   842                                         Dead Space      PS3  2008.0   

      Genre             Pub

## Análisis
1. Filas donde las ventas son mayores a 1 millón en América del Norte [***895*** rows x 11 columns] de [***16598*** rows x 11 columns]
2. Filas donde las ventas son menores a 0.1 millón en Japón [***14021*** rows x 11 columns] de [***16598*** rows x 11 columns]
3. Filas donde el género Acción tuvo ventas globales mayores a 2 millones [***175*** rows x 11 columns] de [***16598*** rows x 11 columns]


# 6. Slicing de Datos
- Selecciona y muestra solo las columnas Name y Global_Sales del DataFrame.
- Utilizando loc[], selecciona y muestra las filas de la 5 a la 10 (inclusive) y las columnas Name y Genre.
- Utilizando iloc[], selecciona y muestra las primeras 5 filas y las primeras 3 columnas del DataFrame.

In [36]:
# Seleccionar columnas por nombre y ventas globales
name_global_sales = df[['Name', 'Global_Sales']]
print(name_global_sales)

# Seleccionar filas y columnas específicas con loc
loc_result = df.loc[4:9, ['Name', 'Genre']]
print("\n\n")
print(loc_result)
# Al contar la fila con índice 0 como la primera fila, se debe entonces bajar un índice, de [5:10] a [4:9]

# Seleccionar filas y columnas específicas con iloc
iloc_result = df.iloc[:5, :3]
print("\n\n")
print(iloc_result)

                                                   Name  Global_Sales
0                                            Wii Sports         82.74
1                                     Super Mario Bros.         40.24
2                                        Mario Kart Wii         35.82
3                                     Wii Sports Resort         33.00
4                              Pokemon Red/Pokemon Blue         31.37
...                                                 ...           ...
16593                Woody Woodpecker in Crazy Castle 5          0.01
16594                     Men in Black II: Alien Escape          0.01
16595  SCORE International Baja 1000: The Official Game          0.01
16596                                        Know How 2          0.01
16597                                  Spirits & Spells          0.01

[16598 rows x 2 columns]



                        Name         Genre
4   Pokemon Red/Pokemon Blue  Role-Playing
5                     Tetris        Puzzle
6 